Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,25 +1,101 @@
-function setAlarm() {}
+const audio = new Audio("alarmsound.mp3");
+let countdownInterval;
+let remainingTime = 0;
 
-// DO NOT EDIT BELOW HERE
+function formatTime(seconds) {
+  const minutes = Math.floor(seconds / 60);
+  const secs = seconds % 60;
+  return `${String(minutes).padStart(2, "0")}:${String(secs).padStart(2, "0")}`;
+}
 
-var audio = new Audio("alarmsound.mp3");
+function updateDisplay(time) {
+  const timeRemaining = document.getElementById("timeRemaining");
+  timeRemaining.innerText = `Time Remaining: ${formatTime(time)}`;
+}
 
-function setup() {
-  document.getElementById("set").addEventListener("click", () => {
-    setAlarm();
-  });
+function setAlarm() {
+  const inputField = document.getElementById("alarmSet");
+  remainingTime = parseInt(inputField.value, 10);
+
+  if (isNaN(remainingTime) || remainingTime <= 0) {
+    alert("Please enter a valid number greater than 0.");
+    return;
+  }
+
+  updateDisplay(remainingTime);
+
+  clearInterval(countdownInterval);
 
-  document.getElementById("stop").addEventListener("click", () => {
-    pauseAlarm();
-  });
+  countdownInterval = setInterval(() => {
+    remainingTime--;
+
+    if (remainingTime >= 0) {
+      updateDisplay(remainingTime);
+    }
+
+    if (remainingTime === 0) {
+      playAlarm();
+      clearInterval(countdownInterval);
+      flashBackground();
+    }
+  }, 1000);
+}
+
+function stopAlarm() {
+  clearInterval(countdownInterval);
+  audio.pause();
+  resetBackground();
 }
 
 function playAlarm() {
   audio.play();
 }
 
-function pauseAlarm() {
-  audio.pause();
+function flashBackground() {
```

Comment:
This is a great addition and a good use of setInterval. There is a bit of bug when I run this though. Clicking the stop alarm button doesn't actually reset the background as intended. I think this is because the interval you set in this function is still going.

You probably need a function to use `clearInterval` like you have done for your `stopAlarm` function for this to work as intended